contracts/src/PDPService.sol

```diff
@@ -184,6 +197,11 @@ contract PDPService {
             totalDelta += removeOneRoot(setId, rootIds[i]);
         }
         proofSetLeafCount[setId] -= totalDelta;
+        // Clear next challenge epoch if the set is now empty.
+        // It will be re-set when new data is added.
+        if (proofSetLeafCount[setId] == 0) {
```

Comment:
I think we should keep a counter of live roots and consult that both for the needsChallengeEpoch bool above and for this one.  I think it makes sense to do here.  But if you'd rather defer that that's ok.

Reply:
I'm not convinced yet, so will defer.